app/stac_api/validators.py

```diff
@@ -216,3 +223,306 @@ def validate_asset_multihash(value):
+            message = f"The application could not decode the query parameter" \
+                      f"Please check the syntax ({error})." \
+                      f"{query}"
+            raise RestValidationError(code='query-invalid', detail=_(message))
+
+        self._query_validate_length_of_query(json_query)
+        for attribute in json_query:
+            if not attribute in queriabel_fields:
+                message = f"Invalid field in query argument. The field {attribute} is not " \
+                          f"a propertie. Use one of these {queriabel_fields}"
+                self.errors[f"query-attributes-{attribute}"] = _(message)
+
+            # validate operators
+            self._query_validate_operators(json_query, attribute)
+
+    def _query_validate_length_of_query(self, json_query):
+        '''Test the maximal number of attributes in the query parameter
+        Args:
+            json_query: dictionary
+                To test how many query attributes are provided
+        Raise:
+            ValidationError
+        '''
+        if len(json_query) > self.max_query_attributes:
+            message = f"Too many attributes in query parameter. Max. " \
+                      f"{self.max_query_attributes} allowed"
+            logger.error(message)
+            raise RestValidationError(code='query-invalid', detail=_(message))
+
+    def _query_validate_operators(self, json_query, attribute):
+        '''
+        Checks if the query param is build out of valid operators (like eq, lt).
+
+        There is a distinction between the allowed operators between string operators
+        and number operators.
+
+        Args:
+            json_query: dict
+                A dictionary of request payload
+            attribute: string
+                The attribute that is filtered by the operator
+        '''
+        int_operators = ["eq", "neq", "lt", "lte", "gt", "gte"]
+        str_operators = ["startsWith", "endsWith", "contains", "in"]
+        operators = int_operators + str_operators
+
+        # iterate trough the operators
+        for operator in json_query[attribute]:
+            if operator in operators:
+                # get the values which shall be filtered with the operator
+                value = json_query[attribute][operator]
+                # validate type to operation
+                if (
+                    isinstance(value, str) and operator in int_operators and
+                    attribute in int_operators
+                ):
+                    message = f"You are not allowed to compare a string/date ({attribute})"\
+                              f" with a number operator." \
+                              f"for string use one of these {str_operators}"
+                    self.errors[f"query-operator-{operator}"] = _(message)
+                if (
+                    isinstance(value, int) and operator in str_operators and
+                    operator in str_operators
```

Comment:
`operator in str_operators` is redundant here. Should one of the two perhaps be `attribute in str_operators` here?

Reply:
will have a look